src/test/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -1169,4 +1170,86 @@ public void testReadStringGetReadLength(final String readBases, final int readLe
     public void testHasAttribute(final SAMRecord samRecord, final String tag, final boolean expectedHasAttribute) {
         Assert.assertEquals(samRecord.hasAttribute(tag), expectedHasAttribute);
     }
+
+    @Test
+    public void test_setAttribute_empty_array() {
+        final SAMFileHeader header = new SAMFileHeader();
+        final String arrayTag = "XA";
+        final SAMRecord record = new SAMRecord(header);
+        Assert.assertNull(record.getStringAttribute(arrayTag));
+        record.setAttribute(arrayTag, new int[0]);
+        Assert.assertNotNull(record.getSignedIntArrayAttribute(arrayTag));
+        Assert.assertEquals(record.getSignedIntArrayAttribute(arrayTag),new int[0]);
+        record.getSAMString();
+        Assert.assertEquals(record.getAttribute(arrayTag), new char[0]);
+        record.setAttribute(arrayTag, null);
+        Assert.assertNull(record.getStringAttribute(arrayTag));
+
+    }
+
+    @DataProvider
+    public Object[][] getEmptyArrays(){
+        final String BAM = BamFileIoUtils.BAM_FILE_EXTENSION;
+        final String SAM = IOUtil.SAM_FILE_EXTENSION;
+        final String CRAM = CramIO.CRAM_FILE_EXTENSION;
+        return new Object[][]{
+                {new int[0], int[].class, BAM},
+                {new short[0], short[].class, BAM},
+                {new byte[0], byte[].class, BAM},
+                {new float[0], float[].class, BAM},
+
+                {new int[0], int[].class, SAM},
+                {new short[0], short[].class, SAM},
+                {new byte[0], byte[].class, SAM},
+                {new float[0], float[].class, SAM},
+
+                {new int[0], int[].class, CRAM},
+                {new short[0], short[].class, CRAM},
+                {new byte[0], byte[].class, CRAM},
+                {new float[0], float[].class, CRAM},
+
+        };
+    }
+
+    @Test(dataProvider = "getEmptyArrays")
+    public void testWriteSamWithEmptyArray(Object emptyArray, Class<?> arrayClass, String fileExtension) throws IOException {
+        Assert.assertEquals(emptyArray.getClass(), arrayClass);
+        Assert.assertEquals(Array.getLength(emptyArray), 0);
+
+        final String arrayTag = "XA";
```

Comment:
might as well get into the habit of using lowercase tags for non-standard ones.

Reply:
good call